ApplicationCode/Commands/OctaveScriptCommands/RicExecuteScriptForCasesFeature.cpp

```diff
@@ -61,30 +62,51 @@ void RicExecuteScriptForCasesFeature::onActionTriggered( bool isChecked )
     RiuMainWindow* mainWindow = RiuMainWindow::instance();
     mainWindow->showProcessMonitorDockPanel();
 
-    RiaApplication* app        = RiaApplication::instance();
-    QString         octavePath = app->octavePath();
-    if ( !octavePath.isEmpty() )
-    {
-        QStringList arguments = RimCalcScript::createCommandLineArguments( scriptAbsolutePath );
+    RiaApplication* app = RiaApplication::instance();
 
-        std::vector<RimCase*> selection;
-        caf::SelectionManager::instance()->objectsByType( &selection );
+    QString             pathToScriptExecutable;
+    QProcessEnvironment processEnvironment;
 
-        // Get case ID from selected cases in selection model
-        std::vector<int> caseIdsInSelection;
-        for ( size_t i = 0; i < selection.size(); i++ )
+    if ( scriptAbsolutePath.endsWith( ".py" ) )
+    {
+        processEnvironment     = app->pythonProcessEnvironment();
+        pathToScriptExecutable = app->pythonPath();
+
+        if ( pathToScriptExecutable.isEmpty() )
         {
-            RimCase* casePtr = selection[i];
-            caseIdsInSelection.push_back( casePtr->caseId );
+            RiaLogging::warning( "Path to Python executable is empty, not able to execute script" );
         }
+    }
+    else
+    {
+        processEnvironment     = app->octaveProcessEnvironment();
+        pathToScriptExecutable = app->octavePath();
+        if ( pathToScriptExecutable.isEmpty() )
+        {
+            RiaLogging::warning( "Path to Octave executable is empty, not able to execute script" );
+        }
+    }
 
-        if ( caseIdsInSelection.size() > 0 )
+    if ( !pathToScriptExecutable.isEmpty() )
+    {
+        QStringList      arguments = RimCalcScript::createCommandLineArguments( scriptAbsolutePath );
+        std::vector<int> caseIdsInSelection;
         {
-            RiaApplication::instance()->launchProcessForMultipleCases( octavePath,
-                                                                       arguments,
-                                                                       caseIdsInSelection,
-                                                                       app->octaveProcessEnvironment() );
+            std::vector<RimCase*> selection;
+            caf::SelectionManager::instance()->objectsByType( &selection );
+
+            // Get case ID from selected cases in selection model
+            for ( size_t i = 0; i < selection.size(); i++ )
```

Comment:
Nitpicking: Looks like index 'i' is not needed, so will look nicer with "for ( RimCase* case : selection )"